lib/plugins/aws/lib/validate.js

```diff
@@ -20,42 +20,25 @@ module.exports = {
       !['logs', 'info', 'deploy', 'remove', 'rollback', 'metrics', 'invoke'].includes(
         this.serverless.processedInput.commands[0]
       ) ||
-      this.serverless.processedInput.commands[1] === 'local'
+      this.serverless.processedInput.commands[1] === 'local' ||
+      (this.serverless.processedInput.options &&
+        this.serverless.processedInput.options.ec2Credentials)
     ) {
       return BbPromise.resolve();
     }
+
     const creds = Object.assign({}, this.provider.getCredentials());
     delete creds.region;
     delete creds.signatureVersion;
 
-    if (
-      Object.keys(creds).length === 0 &&
-      !process.env.ECS_CONTAINER_METADATA_FILE &&
-      !process.env.AWS_CONTAINER_CREDENTIALS_RELATIVE_URI &&
-      !process.env.AWS_CONTAINER_CREDENTIALS_FULL_URI
-    ) {
-      // first check if the EC2 Metadata Service has creds before throwing error
-      const metadataService = new this.provider.sdk.MetadataService({
-        httpOptions: { timeout: 100, connectTimeout: 100 }, // .1 second timeout
-        maxRetries: 0, // retry 0 times
-      });
-      return new BbPromise((resolve, reject) =>
-        metadataService.request('/', (err, data) => {
-          return err ? reject(err) : resolve(data);
-        })
-      )
-        .catch(() => null)
-        .then(identity => {
-          if (!identity) {
-            const message = [
-              'AWS provider credentials not found.',
-              ' Learn how to set up AWS provider credentials',
-              ` in our docs here: <${chalk.green('http://slss.io/aws-creds-setup')}>.`,
-            ].join('');
-            userStats.track('user_awsCredentialsNotFound');
-            throw new this.serverless.classes.Error(message);
-          }
-        });
+    if (Object.keys(creds).length === 0) {
+      const message = [
```

Comment:
Maybe we can provide a meaningful message for EC2 users (by detecting `ECS_CONTAINER_METADATA_FILE` env var), so they should use `ec2Credentials` options.

Is `ec2Credentials` an option that's send manually by the user? Is it read by AWS? Where is it documented exactly?

Reply:
the `ec2Credentials` is an option passed to serverless command, I've added some words to `credentials.md` in order to explain it. I hope it helps.